added example to terraform state list documentation (#35377)
hashicorp/terraform: website/docs/cli/commands/state/list.mdx

aws_instance.bar[1]
```

> ## Example: Filtering with Index

Reviewer: ```suggestion
## Example: Filtering with an index
```
```suggestion
## Example: Filtering with Index
```

> This example will show you how to filter when your filter includes an Index:

Reviewer: ```suggestion
This example will show you how to filter when your filter includes an index:
```

```
$ terraform state list 'foo[0].bar'
foo[0].bar.lorem
foo[0].bar.ipsum
```

> When there's an index in the filter then you have to use single quotes.

Reviewer: ```suggestion
Use single quotes to prevent your command line from interpreting the square brackets as shell syntax.
```
